### PR TITLE
Season class

### PR DESCRIPTION
**Pull request by AlexisDumortier**
The rational behind implementing a season class is to group in one class all the methods relevant to a specific season.

Changes involved in this PR :
- a season class that includes the methods relevant to a season 
- tests for that class
- game_collection is loaded in initialize until we find a better way of doing that
- the fixture file is loaded when running our test and the full csv file is loaded when running spec harness

### CODE REVIEW

File: lib/game_collection.rb [hunk 1]
```diff
@@ -73,74 +77,6 @@ def games_per_season
     @games_per_season ||= @games.group_by{|game| game.season}
   end
 
-  def teams_for_season(season_id)
```
Comment (AlexisDumortier) [inline lib/game_collection.rb]: moved all these methods to the season class

File: lib/season.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
+require_relative './game_collection'
+require_relative './game_teams'
+
+class Season
+
+  def initialize(csv_file_path = './test/fixtures/games.csv')
```
Comment (AlexisDumortier) [inline lib/season.rb]: the Season methods require a GameCollection (would be ideal not to have to initialize the GameCollection in the season class though)

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -23,6 +23,7 @@ def initialize(game_path, team_path, game_teams_path)
     @game_teams = GameTeams.from_csv(game_teams_path)
     Team.from_csv(team_path)
     @teams = Team.all
+    @season = Season.new(game_path)
```
Comment (AlexisDumortier) [inline lib/stat_tracker.rb]: when a game_path is not specified, the season is initialized with the fixture file game_path

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -66,19 +67,19 @@ def count_of_games_by_season
   end
 
   def winningest_coach(season_id)
-    @game_collection.winningest_coach(season_id)
+    @season.winningest_coach(season_id)
```
Comment (AlexisDumortier) [inline lib/stat_tracker.rb]: the methods relevant to the season are now called on season objects

File: test/game_collection_test.rb [hunk 5]
```diff
@@ -49,55 +49,28 @@ def test_it_can_return_percentage_tie_wins
   end
 
   def test_count_of_games_by_season_method
-    expected = {"20122013"=>12, "20152016"=>9, "20132014"=>1, "20142015"=>3, "20162017"=>1}
+    expected = {'20122013'=>12, '20152016'=>9, '20132014'=>1, '20142015'=>3, '20162017'=>1}
```
Comment (AlexisDumortier) [inline test/game_collection_test.rb]: changed to single quotes (apparently it's preferred in Ruby when string interpolation is not required)

Comment (DavidHoltkamp1) [reply test/game_collection_test.rb]: What, really?!!

**Comment by palworth** [comment]
Creating a season class to improve OOP. Some of our methods were using multiple data structures from different CSV files, the season class houses these methods so our game, team, and game_team  classes have more single responsibility.

**Pull request merged**